x/auth/ante/tx_priority.go

```diff
@@ -0,0 +1,55 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// txPriorityDecorator implements tx antehandler that determines a
+// transaction's priority via a naive mechanism -- the total sum of fees provided.
+// It sets the Priority in CheckTx only.
+//
+// We set the Priority of the transaction
+// to be ordered in the Tendermint mempool based naively on the total sum of all
+// fees included. Applications that need more sophisticated mempool ordering
+// should look to implement their own fee handling middleware instead of using
+// TxPriorityHandler.
+type txPriorityDecorator struct{}
+
+func NewTxPriorityDecorator() sdk.AnteDecorator {
+	return txPriorityDecorator{}
+}
+
+// AnteHandle implements sdk.AnteDecorator. We set the Priority of the transaction
+// to be ordered in the Tendermint mempool based naively on the total sum of all
+// fees included. Applications that need more sophisticated mempool ordering
+// should look to implement their own fee handling middleware instead of using
+// TxPriorityHandler.
+func (vbd txPriorityDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !ctx.IsCheckTx() {
+		return next(ctx, tx, simulate)
+	}
+
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	feeCoins := feeTx.GetFee()
+
+	// TODO What's the best way to set the priority on the response?
+	GetTxPriority(feeCoins)
```

Comment:
@alexanderbez The tx priority setter should go in ante handlers, right? What's the best way you can think of to set them in an antehandler and populate back on CheckTxResponse?

It seems like changing the antehandler signature function would work:
```go
AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, priority, error) {
```

But it doesn't seem satisfactory: it's not used in DeliverTx or Simulate, and it's a big API-breaking change.

Reply:
Right, so for the sake of getting this out and making sure we don't introduce substantial _new_ changes when reverting to the old method, what I would recommend is that we actually set the `TxPriority` on the `sdk.Context`. So in the AnteHandler, we do `ctx = ctx.WithTxPriority(txPriority)` and when we return that `ctx`, `CheckTx` will know what to return.